### DISCUSSION TITLE
Device configuration file problem

### DISCUSSION THREAD

**Discussion by HolBaum5** [zwave-js/zwave-js-ui#3590] [Q&A] [answered]
Hello, I have a Z-TRV-V01 thermostatic valve whose config is not (yet) in the database. 
I have put the config as described here (https://community.home-assistant.io/t/how-to-manualy-supply-z-wave-js-config-file-from-the-db/424393/9) to store/config.
![grafik](https://github.com/zwave-js/node-zwave-js/assets/79714761/cf6c7005-46a1-493b-b611-f0e26763c621)
 
I have taken the config file from here https://github.com/zwave-js/node-zwave-js/pull/6542

But unfortunately I don't see any change in my overview - the device is still unknown. What could be the reason?
![grafik](https://github.com/zwave-js/node-zwave-js/assets/79714761/ff1adb63-f201-4e22-bad3-df1a18e4949e)

**Comment by robertsLando** [comment]
@HolBaum5 Try to re-interview your device. Also check in logs if the configuration is picked up correctly

**Comment by kpine** [comment]
First, make sure you restart Z-Wave JS UI so the config file is loaded. Second, the config file you've linked to does not match your device. Here are the PR values:

```
"manufacturerId": "0x045a"
"productType": "0x0000",
"productId": "0x0400"
```

Your `productType` is `0x0501`, so the config will not be matched. If these are the same devices, either the PR you've linked to is incorrect, or there are multiple versions of the same device. For the former case, just change `productType` from `0x0000` to `0x0501`. For the latter case, you'd need to append your variation:

```json
	"devices": [
		{
			"productType": "0x0000",
			"productId": "0x0400"
		},
		{
			"productType": "0x0501",
			"productId": "0x0400"
		}
	],
```

Given the author of the PR says "I'm not sure which product type it is.", it might just be incorrect. It doesn't sound like they are actual using it this device?

**Reply by HolBaum5** [reply]
Thanks for the tip I will try the change and get back to you.

**Comment by HolBaum5** [comment]
This is what I read from the log file (below). Therefore, I think devices must be 

```
	"devices": [
		{
			"productType": "0x0400",
			"productId": "0x0501"
		}
	],
```
[0x0501.json](https://github.com/zwave-js/zwave-js-ui/files/14232104/0x0501.json)


However, this does not seem to work. 
I tried with productType = 0x0501 and productId = 0x0400.  
I restart the Docker Z-Wave container every time. 
How can I check whether my configuration file is being used and is correct?

```
2024-02-10T08:08:42.232Z SERIAL « 0x011600a80000010016087205045a0400050100a8007f7fdf                  (24 bytes)
2024-02-10T08:08:42.234Z SERIAL » [ACK]                                                                   (0x06)
2024-02-10T08:08:42.236Z CNTRLR   [Node 022] [Manufacturer Specific] manufacturerId: metadata updat [Endpoint 0]
                                  ed
2024-02-10T08:08:42.242Z CNTRLR   [Node 022] [+] [Manufacturer Specific] manufacturerId: 1114       [Endpoint 0]
2024-02-10T08:08:42.245Z CNTRLR   [Node 022] [Manufacturer Specific] productType: metadata updated  [Endpoint 0]
2024-02-10T08:08:42.247Z CNTRLR   [Node 022] [+] [Manufacturer Specific] productType: 1024          [Endpoint 0]
2024-02-10T08:08:42.249Z CNTRLR   [Node 022] [Manufacturer Specific] productId: metadata updated    [Endpoint 0]
2024-02-10T08:08:42.251Z CNTRLR   [Node 022] [+] [Manufacturer Specific] productId: 1281            [Endpoint 0]
2024-02-10T08:08:42.253Z DRIVER « [Node 022] [REQ] [BridgeApplicationCommand]
                                  │ RSSI: -88 dBm
                                  └─[ManufacturerSpecificCCReport]
                                      manufacturer id: 0x045a
                                      product type:    0x0400
                                      product id:      0x0501
2024-02-10T08:08:42.255Z CNTRLR « [Node 022] received response for manufacturer information:
                                    manufacturer: unknown (0x045a)
                                    product type: 0x0400
                                    product id:   0x0501
```

**Reply by kpine** [reply, marked as answer]
Look at the start of the logs for some error.

**Comment by apella12** [comment]
I'm not an expert like the other commentators, but you haven't copied the file from the PR correctly (in the 501.json).  IMO the $import needs to be the first line like in the PR.  My lessor concern is the lack of a `templates` in the config folder, but I think `~/templates/` is okay based on what I have done in the past.  However, if there are $import references as `/templates/` those need to be added. Don't mess with anything in the PR except the Type:ID.

**Comment by HolBaum5** [comment]
Thanks for the hint with the log file, the following errors (see below) appeared there. After I corrected them, the device is displayed correctly. I will add the information in the PR as info.

![grafik](https://github.com/zwave-js/zwave-js-ui/assets/79714761/b827e633-6031-4c2d-b194-2015dc8a9f89)


```
2024-02-13T06:51:58.111Z CONFIG   Error parsing config file 0x0501.json: The devices config file is malformed!
                                  packages/config/config/devices/0x0501.json:
                                  Parameter #13 is missing defaultValue, which is required unless the parameter 
                                  is readOnly (ZW0340)

2024-02-13T07:03:59.392Z CONFIG   Error parsing config file 0x0501.json: The devices config file is malformed!
                                  packages/config/config/devices/0x0501.json:
                                  Parameter #14 is missing defaultValue, which is required unless the parameter 
                                  is readOnly (ZW0340)
								  
2024-02-13T07:10:10.838Z CONFIG   Error parsing config file 0x0501.json: The devices config file is malformed!
                                  packages/config/config/devices/0x0501.json:
                                  Parameter #4 is missing required property "minValue"! (ZW0340)

2024-02-13T07:16:21.793Z CONFIG   Error parsing config file 0x0501.json: The devices config file is malformed!
                                  packages/config/config/devices/0x0501.json:
                                  Parameter #4 is missing required property "maxValue"! (ZW0340)								  

```

**Reply by kpine** [reply]
Your file is not the same as the PR file. Seems like you downloaded the wrong version.

**Reply by HolBaum5** [reply]
Yes, you're right, I took the wrong version, sorry.
